common/djangoapps/third_party_auth/tests/test_samlproviderconfig.py

```diff
@@ -0,0 +1,17 @@
+import unittest
+
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+from third_party_auth.models import SAMLProviderConfig
+
+class SAMLProviderConfigTests(APITestCase):
+    """
+        API Tests for SAMLProviderConfig REST endpoints
+    """
+    def test_fetch_one_config(self):
+        # ^^auth/samlproviderconfig/(?P<pk>[^/.]+)/$ [name='samlproviderconfig-detail']
+        url = reverse('samlproviderconfig-detail')
```

Comment:
ie `reverse('samlproviderconfig-detail', pk=1)`

Reply:
Thanks @raq929 

I switched to using a ModelViewSet since that seems to contain all the functionality I am looking for basic CRUd with auth support for models,

But I am still stuck in the testing side:

Actually `reverse('samlproviderconfig-detail', pk=1)` failed with ```        url = reverse('samlproviderconfig-detail', args=[1])```

I also tried ```        url = reverse('samlproviderconfig-detail', args=[1])``` that did not work
I also tried ```        url = reverse('samlproviderconfig-detail', args=['1'])``` did not work
also tried ```        url = reverse('samlproviderconfig-detail', kwargs={'pk':1})``` that also did not work

I was pretty sure that last one was the correct, based on the output shown below:

None of these worked for me

I get this failure with the `args` version

```
E       django.urls.exceptions.NoReverseMatch: Reverse for 'samlproviderconfig-detail' with arguments '(1,)' not found. 2 pattern(s) tried: ['^auth/saml/v0/providerconfig/(?P<pk>[^/.]+)\\.(?P<format>[a-z0-9]+)/?$', '^auth/saml/v0/providerconfig/(?P<pk>[^/.]+)/$']
```

so it is picking up my url mappings but reverse somehow is still not liking my mapping name somehow

Reply:
Huh, I feel like that last one should work. Have you tried printing out the output of `reverse`? That might shed some insight as to what it's doing. wrong.
Your mapping name seems good to me. Unless it's already used elsewhere. Or somehow isn't registered? But then it wouldn't work manually either. Let me know if you want to get on a call sometime, I'm pretty good at convincing Django to do what I want it to do.

Reply:
this part is now resolved, turns out using leading caret was the main issue ... thanks @raq929 

now trying to figure out how to test with some mock data.. so I don't just test 404 case